# Desafío 1: Hola Mundo

   ● Utilice el método print para mostrar en pantalla la siguiente línea:
   
Hola Mundo, esta es mi primera incursión en Python


In [1]:
print('Hola Mundo, esta es mi primera incursión en Python')

Hola Mundo, esta es mi primera incursión en Python


# Desafío 2: Ingreso de variables

Ingrese una serie de datos respecto a su información personal. Esta información debe estar
ingresada en variables para su posterior uso. Ingrese los siguientes datos:

    ● Su nombre.
    ● Edad.
    ● Una lista con sus 3 actividades favoritas.
    ● ¿Tiene mascotas?

In [2]:
nombre = 'Paula Cristina Navarrete Campos'
edad = 37
actividades = ['treking', 'ajedrez' , 'leer']
mascotas = True


# Desafío 3: Imprimiendo Variables

    ● Imprima cada variable creada.
        ○ Acceda al segundo elemento de su lista.
    ● Pida los tipos de datos asociados.
    ● Interpole los datos en un método print para generar una presentación sobre usted.


In [3]:
print('Nombre:',nombre)
print('Edad:',edad)
print('Actividad Favorita:',actividades[1])
print('¿Tiene mascotas?:', 'Si' if mascotas else 'No')

Nombre: Paula Cristina Navarrete Campos
Edad: 37
Actividad Favorita: ajedrez
¿Tiene mascotas?: Si


In [4]:
print('El tipo de datos de la variable nombre es:',type(nombre))
print('El tipo de datos de la variable edad:',type(edad))
print('El tipo de datos de la variable actividades Favorita:',type(actividades))
print('El tipo de datos de la variable mascotas:', type(mascotas))

El tipo de datos de la variable nombre es: <class 'str'>
El tipo de datos de la variable edad: <class 'int'>
El tipo de datos de la variable actividades Favorita: <class 'list'>
El tipo de datos de la variable mascotas: <class 'bool'>


In [5]:
print('Hola! Mi nombre es {}, tengo {} años,{} mis actividades favoritas son {}, {} y {}.'
      .format(nombre, edad, mascotas*' me encanta pasar tiempo con mis mascotas,', actividades[0],actividades[1], actividades[2]))

Hola! Mi nombre es Paula Cristina Navarrete Campos, tengo 37 años, me encanta pasar tiempo con mis mascotas, mis actividades favoritas son treking, ajedrez y leer.


# Desafío 4

A continuación se presenta una serie de códigos que generan errores. Corríjalos y comente
cuáles fueron las fallas.


## 4.1

In [6]:
print('Estaba la pájara pinta sentada en el verde limón)

SyntaxError: EOL while scanning string literal (Temp/ipykernel_22356/1236003848.py, line 1)

### Rp: Error de sintaxis, falla porque falta el cierre de comillas de término de string antes del cierre del paréntesis del print.

In [7]:
print('Estaba la pájara pinta sentada en el verde limón')

Estaba la pájara pinta sentada en el verde limón


## 4.2

In [8]:
print('Mi nombre es' name 'y tengo' edad, 'años')

SyntaxError: invalid syntax (Temp/ipykernel_22356/1605563620.py, line 1)

In [9]:
name=nombre
print('Mi nombre es', name, 'y tengo', edad, 'años')


Mi nombre es Paula Cristina Navarrete Campos y tengo 37 años


### Rp: Error de sintaxis, falla porque falta insertar coma antes y después de las variables name y edad antes en el output del print.

## 4.3

In [10]:
import padnas as pd
import nunnpy as np

ModuleNotFoundError: No module named 'padnas'

In [11]:
import pandas as pd
import numpy as np


### Rp: Falla porque no encuentra el módulo 'padnas' o 'nunnpy', porque no existen con ese nombre, ya que se encuentan mal escritos.

## 4.4

In [12]:
"Ornitorrinco" + 45

TypeError: can only concatenate str (not "int") to str

In [13]:
"Ornitorrinco" + '45'

'Ornitorrinco45'

### Rp. Error de tipo de datos, falla porque no puede sumar números con palabras. Para evitarlo habría que homogeneizar los tipos de datos convirtiendo a string el número 45.

# Desafío 5: Utilizando pandas y numpy

## 5.1 Importe pandas y numpy.


In [14]:
import pandas as pd
import numpy as np

## 5.2  Posteriormente importe el archivo flights.csv.

In [15]:
df = pd.read_csv('flights.csv')

## 5.3 Solicite las primeras y últimas 5 observaciones de la tabla de datos.



In [16]:
print(df.head(5))
print(df.tail(5))

   year     month  passengers
0  1949   January         112
1  1949  February         118
2  1949     March         132
3  1949     April         129
4  1949       May         121
     year      month  passengers
139  1960     August         606
140  1960  September         508
141  1960    October         461
142  1960   November         390
143  1960   December         432


## 5.4 Utilice describe para obtener las medidas univariadas de la columna year.

In [17]:
(df['year'].describe())

count     144.000000
mean     1954.500000
std         3.464102
min      1949.000000
25%      1951.750000
50%      1954.500000
75%      1957.250000
max      1960.000000
Name: year, dtype: float64

## 5.5 Utilice value_counts para obtener la frecuencia de meses y años de la tabla.

In [18]:
print(df['year'].value_counts())
print(df['month'].value_counts())

1949    12
1950    12
1951    12
1952    12
1953    12
1954    12
1955    12
1956    12
1957    12
1958    12
1959    12
1960    12
Name: year, dtype: int64
January      12
February     12
March        12
April        12
May          12
June         12
July         12
August       12
September    12
October      12
November     12
December     12
Name: month, dtype: int64


## 5.6 Guarde en variables las primeras y últimas 15 observaciones de la tabla.


In [19]:
head = df.head(5)
tail = df.tail(15)

## 5.7 Solicite la media, mediana y desviación estándar de la cantidad de pasajeros utilizando los métodos de numpy. Replique el procedimiento para las últimas y primeras 15 observaciones. Comente brevemente los resultados.


In [20]:
passengers_mean = np.mean(df['passengers'])
passengers_median = np.median(df['passengers'])
passengers_std = np.std(df['passengers'])

print(passengers_mean,passengers_median,passengers_std)

280.2986111111111 265.5 119.54904153277148


Podemos ver que el promedio de pasajeros por vuelo para la muestra es de aproximadamente 280. La mediana corresponde a 265 pasajeros, menor a la media, por lo que más de la mitad de los vuelos fueron realizados con menos pasajeros que la media. La desviación estándar es de aproximadamente 120 pasajeros, esto implica que la gran mayoría de los vuelos transportó más de 160 pasajeros y menos de 400.